src/main/java/org/broadinstitute/consent/http/models/User.java

```diff
@@ -279,7 +279,7 @@ public void addRole(UserRole userRole) {
             this.setRoles(new ArrayList<>());
         }
 
-        if (!this.getRoles().stream().map(UserRole::getRoleId).collect(Collectors.toList()).contains(userRole.getRoleId())) {
+        if (!this.getRoles().contains(userRole)) {
```

Comment:
this is so sneaky, wow. I think I wrote this condition a while back, impressive catch.

Reply:
I've recently started moving away from list `contains` due to performance issues compared to hash `contains`. It should not be an issue here, just pointing it out.